prov/util/src/util_mr_cache.c

```diff
@@ -399,6 +400,12 @@ struct ofi_mr_entry *ofi_mr_cache_find(struct ofi_mr_cache *cache,
 		goto unlock;
 	}
 
+	monitor = cache->monitors[entry->info.iface];
+	if (!monitor->valid(monitor, &entry->info, entry)) {
+		entry = NULL;
```

Comment:
Should the invalid entry be purged here?

Reply:
It should. Will add.